stress_test/nb_generator.py

```diff
@@ -0,0 +1,254 @@
+                          format(oper_ds_found_flows))
+            if (oper_ds_found_flows - previous_discovered_flows) != 0:
+                t_discovery_start = time.time()
+                previous_discovered_flows = oper_ds_found_flows
+            if oper_ds_found_flows == self.total_flows:
+                time_interval = time.time() - t_start
+                logging.debug('[NB_generator] Flow-Master '
+                              '{0} flows found in {1} seconds'.
+                              format(self.total_flows, time_interval))
+                result_queue.put({'confirm_time': time_interval}, block=True)
+                return
+        time.sleep(1)
+
+
+def poll_flows_switches(self, result_queue, t_start):
+    """
+    Monitors operational flows in switches of Multinet until the expected
+    number of flows are found or the deadline is reached.
+    :param t_start: timestamp for begin of discovery
+    :returns: Returns a float number containing the time in which
+    total flows were discovered otherwise containing -1.0 on failure.
+    :rtype: float
+    :type t_start: float
+    """
+    deadline = 240
+    t_discovery_start = time.time()
+    previous_discovered_flows = 0
+    while True:
+        if (time.time() - t_discovery_start) > deadline:
+            logging.info('[NB_generator] Deadline of {0} seconds '
+                         'passed'.format(deadline))
+            result_queue.put({'switch_operation_time': -1.0}, block=True)
+            return
+        else:
+            discovered_flows = self.multinet.get_flows()
+            logging.debug('[NB_generator] Found {0} flows at topology switches'
+                          .format(discovered_flows))
+            if (discovered_flows - previous_discovered_flows) != 0:
+                t_discovery_start = time.time()
+                previous_discovered_flows = discovered_flows
+            if discovered_flows == self.total_flows:
+                time_interval = time.time() - t_start
+                logging.debug('[NB_generator] expected flows = {0} \n '
+                              'discovered flows = {1}'.
+                              format(self.total_flows, discovered_flows))
+                result_queue.put({'switch_operation_time': time_interval},
+                                 block=True)
+                return
+        time.sleep(1)
+    return
+
+
+def monitor_threads_run(self, t_start):
+    """
+    Monitors operational flows in switches of Multinet until the expected
+    number of flows are found or the deadline is reached.
+    :param t_start: timestamp for begin of discovery
+    :returns: Returns a float number containing the time in which
+    total flows were discovered otherwise containing -1.0 on failure.
+    :rtype: float
+    :type t_start:
+    """
+
+    logging.info('[NB_generator] creating result queues for ...')
+    result_queue = multiprocessing.Queue()

```

Comment:
based on a previous comment, this queue is not needed if we store the monitoring threads results within the class